tests/define_test_project.py

```diff
@@ -79,7 +79,7 @@ class _DynamicTestProject(_TestProject):
 @_DynamicTestProject.operation
 @group3
 @_DynamicTestProject.pre.after(op1)
-@_DynamicTestProject.post.true("dynamic")
+@_DynamicTestProject.post(lambda job: job.sp.get("dynamic", False))
```

Comment:
I think this was a bug in the tests. The previous condition `post.true("dynamic")` was checking the job _document_ but the purpose of the dynamic execution is to change the _state point_. This led to incorrect behavior that was exposed by a mismatch in the value `expected_number_of_steps`, set below. I believe I have the correct number of steps now (see comment below).

Reply:
Approved by @b-butler and myself on the other thread.